src/library/scala/collection/generic/IsIterableLike.scala

```diff
@@ -110,10 +113,17 @@ object IsIterableLike {
       val conversion = implicitly[String => IterableOps[Char, Iterable, String]]
     }
 
-  implicit def arrayRepr[T](implicit conv: Array[T] => IterableOps[T, Iterable, Array[T]]): IsIterableLike[Array[T]] =
-    new IsIterableLike[Array[T]] {
-      type A = T
-      val conversion: Array[T] => IterableOps[A, Iterable, Array[T]] = conv
+  implicit def arrayRepr[A0: ClassTag]: IsIterableLike[Array[A0]] { type A = A0 } =
+    new IsIterableLike[Array[A0]] {
+      type A = A0
+      val conversion: Array[A] => IterableOps[A, Iterable, Array[A]] = a => new IterableOps[A, Iterable, Array[A]] {
+        def toIterable: Iterable[A] = ImmutableArray.unsafeWrapArray(a)
```

Comment:
Since we only need a `scala.collection.Iterable` anyway it would be better to use `WrappedArray`. The ownership of the array is not sufficiently clear here to silently wrap it in an `ImmutableArray` IMHO.